sig-robotics/proposal/Definition of cloud robot CRD and Operator/Definition of cloud robot CRD and Operator.md

```diff
@@ -0,0 +1,166 @@
+
+This proposal presents a design solution for a universal CRDs and Operator, enabling resource monitoring and automated control of cloud-native robots. 
+
+## Goals
+
+For developers or end users of CRDs and Operators, the goal of this project is to provide the following capabilities
+
+CRDs are defined to achieve
+
+- Difining robot resources
+- Extending the robot API to enable robot control
+
+Operator is designed to achieve
+
+- automatizing  robotic operations
+- Monitoring robot status
+- automatizing fault handling
+- adaptive controlling
+
+## Proposal
+
+The CRDs and Controllers is designed to provide more flexible and scalable management solutions for cloud robots connected to the KubeEdge platform.
+
+project scope：
+
+The scope of this project includes
+
+- Providing a set of CRDs  and an Operators for managing cloud robotics based on KubeEdge architecture
+  - CRDs to define robot properties and states
+  - CRDs to extend teh robot API, through which we can control the robot
+  - A series of operators to realize the automatic control of the robot
+- This project is to design the robot CRD based on the [interface definition of industrial mobile robots](https://www.ttbz.org.cn/Pdfs/Index/?ftype=st&pms=79882) proposed by the Mobile Robot Industry Alliance
+
+targeting users：
+
+- Smart warehouse manager: As a manager in a real industrial scene, I hope to have a common robot management and control scheme to reduce management and maintenance costs.
+- System user: As a system user, I want to release a task and the system will automatically assign these tasks to specific robot nodes.
+- Robot maintainer: As a robot maintainer, I want the robot to have the ability to handle simple failures, and perform operations such as charging, thereby liberating manpower.
+
+## Design Details
+
+### Architecture
+
+For this project, the architecture adopted basically follows the overall architecture of KubeEdge. The architecture diagram of the project is shown in the figure below:
+
+<img src="./images/architecture.png" alt="architecture" style="zoom:90%;" />
+
+The current predominant approach for managing cloud robots is to treat the robot as a unified entity, considering it as a Device and accessing the KubeEdge platform for management and scheduling purposes.
+
+However, given that a robot is inherently composed of diverse sensors and controllers, it is more appropriate to view it as a combination of heterogeneous components rather than an indivisible arrangement. Consequently, in this solution, a robot is treated as an edge node, with its sensors serving as the device access platform for that node.
```

Comment:
Your first two points are exactly right.

For the third point, Kubeedge manages the accessed devices through two CRDs, `Device` and `Device Instance`, whereas in my CRD definition, the fields related to sensors, are only there to determine that a certain robot is able to perform a specific task, but are not responsible for the management of these sensors.

And the last, when a robot node registers, it reports some necessary information, which includes the sensors it has, and that information will be updated into robot CRD. I hope this might answer your question.